src/gbxml/ForwardTranslator.cpp

```diff
@@ -412,35 +411,36 @@ namespace gbxml {
     return true;
   }
 
-  boost::optional<pugi::xml_node> ForwardTranslator::translateFacility(const openstudio::model::Facility& facility, pugi::xml_node& parent)
+  boost::optional<pugi::xml_node> ForwardTranslator::translateFacility(const openstudio::model::Model& model, pugi::xml_node& parent)
   {
+
+    // `model` is `const`, so we shouldn't call getUniqueModelObject<model::Facility> which will **create** a new object in there.
+    boost::optional<model::Facility> _facility = model.getOptionalUniqueModelObject<model::Facility>();
+
     auto result = parent.append_child("Campus");
-    m_translatedObjects[facility.handle()] = result;
+    std::string name = "Facility";
 
-    boost::optional<std::string> name = facility.name();
+    if (_facility) {
+      m_translatedObjects[_facility->handle()] = result;
+      if (auto _s = _facility->name()) {
+        name = _s.get();
+      }
+    }
```

Comment:
Check if there is an actual explicitly instantiated Facility object to use, otherwise default things.